src/coreclr/tools/cdac-build-tool/ContractReader.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace Microsoft.DotNet.Diagnostics.DataContract.BuildTool;
+
+public partial class ContractReader
+{
+
+    private readonly DataDescriptorModel.Builder _builder;
+
+    public ContractReader(DataDescriptorModel.Builder builder)
+    {
+        _builder = builder;
+    }
+
+    public async Task<bool> ParseContracts(string contractFilePath, CancellationToken token = default)
```

Comment:
Do we need the `bool`? It seems like we are relying on exceptions for the failure case.

Reply:
removed the exceptions.  added line numbers to error messages